src/braket/flake8_plugins/braket_checkstyle_plugin.py

```diff
@@ -278,28 +279,46 @@ def _check_annotation(
         node: ast.FunctionDef,
     ) -> None:
         annotation = None
+        default_value = None
         if arg_type == ArgType.DEFAULT:
             if node.args.args[arg_index].annotation:
                 annotation = node.args.args[arg_index].annotation
+            if node.args.defaults:
+                default_index = arg_index - (len(node.args.args) - len(node.args.defaults))
+                if default_index >= 0 and node.args.defaults[default_index]:
+                    default_value = node.args.defaults[default_index]
         elif arg_type == ArgType.KEYWORD:
             if node.args.kwonlyargs[arg_index].annotation:
                 annotation = node.args.kwonlyargs[arg_index].annotation
+        documented_type = _remove_all_spaces(arg_hint[1:-1])
         if annotation:
-            documented_type = _remove_all_spaces(arg_hint[1:-1])
             annotation_doc = _remove_all_spaces(self._annotation_to_doc_str(annotation))
             if not _are_type_strings_same(annotation_doc, documented_type):
                 self.add_problem(
                     node=node,
                     code="BCS005",
                     arguments=(arg_name, annotation_doc, documented_type),
                 )
+        if (
+            default_value
+            and isinstance(default_value, ast.Constant)
+            and default_value.value is None
+        ):
+            if not documented_type.endswith("|None") and not documented_type.startswith("Optional"):
```

Comment:
Is there a way to do this through attributes without string processing? Seems this allows `str | None` and `Optional[str]`, but what about `Union[str, None]` or `None | str`?

Reply:
I think it makes sense to standardize on putting | None last, and we'll see if we need to deal with edge cases for Union.

Reply:
Actually checking through the types would be better, but it would mean processing them twice or refactoring around how we deal with types. I'm not sure that it's worth it at this point.